Add docker

LeonovichMariia opened the pull request.

LeonovichMariia committed:
feat: добавлен файл docker-compose.yaml, написаны Dockerfile для сервисов — server и gateway.
fix: приложение разбито на два shareIt-server, shareIt-gateway, настроено сообщение между ними через REST.

LeonovichMariia committed:
fix: правки в application.properties

LeonovichMariia committed:
delete: удален класс ValidationMessages в server.
fix: метод delete в UserController не void

LeonovichMariia committed:
fix: правки для починки тестов

LeonovichMariia committed:
fix: правки в ItemServiceImplTest(getPersonal)

KirPyt commented on docker-compose.yml:
```diff
@@ -0,0 +1,36 @@
+      - SPRING_DATASOURCE_URL=jdbc:postgresql://db:5432/share_it
+      - SPRING_DATASOURCE_USERNAME=root
+      - SPRING_DATASOURCE_PASSWORD=root
+#      - TZ=Europe/Moscow
```
Стоит убрать неиспользуемый код

KirPyt commented on gateway/Dockerfile:
```diff
@@ -0,0 +1,4 @@
+FROM amazoncorretto:11-alpine-jdk
+#ENV JAVA_TOOL_OPTIONS -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:8081
```
Стоит убрать неиспользуемый код

KirPyt commented on gateway/src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -0,0 +1,70 @@
+
+    @GetMapping
+    public ResponseEntity<Object> getPersonal(@RequestHeader("X-Sharer-User-Id") long userId,
+                                              @RequestParam(name = "from", defaultValue = "0") @Min(0) Integer from,
```
Рекомендую здесь использовать `@PositiveOrZero`, данная аннотация более точно опишет ограничение

KirPyt commented on gateway/src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -0,0 +1,70 @@
+    @GetMapping("/search")
+    public ResponseEntity<Object> searchItem(@RequestHeader(value = "X-Sharer-User-Id") long userId,
+                                             @RequestParam String text,
+                                             @RequestParam(name = "from", defaultValue = "0") @Min(0) Integer from,
```
Рекомендую здесь использовать `@PositiveOrZero`, данная аннотация более точно опишет ограничение

KirPyt commented on gateway/src/main/java/ru/practicum/shareit/request/ItemRequestController.java:
```diff
@@ -0,0 +1,50 @@
+
+    @GetMapping("all")
+    public ResponseEntity<Object> getOtherUsersRequests(@RequestHeader("X-Sharer-User-Id") Long userId,
+                                                        @RequestParam(name = "from", defaultValue = "0") @Min(0) Integer from,
```
Рекомендую здесь использовать `@PositiveOrZero`, данная аннотация более точно опишет ограничение

KirPyt commented on gateway/src/main/java/ru/practicum/shareit/user/UserController.java:
```diff
@@ -0,0 +1,51 @@
+import javax.validation.Valid;
+
+@Slf4j
+@Validated
```
В данной аннотации здесь нет смысла, так как нет проверяемых аннотаций в параметрах методов

KirPyt commented on server/pom.xml:
```diff
@@ -0,0 +1,91 @@
+
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-validation</artifactId>
```
Валидацию из сервера надо убрать

KirPyt commented on server/src/main/java/ru/practicum/shareit/booking/controller/BookingController.java:
```diff
@@ -10,7 +10,6 @@
-import javax.validation.Valid;
 import javax.validation.constraints.Min;
 import javax.validation.constraints.PositiveOrZero;
 import java.util.List;
```
Из данного класса надо убрать `@Validated` и аннотации в параметрах методов контроллеров, которые валидируют параметры (`@PositiveOrZero` и `@Min(1)`)

KirPyt commented on server/src/main/java/ru/practicum/shareit/booking/dto/BookingCreationDto.java:
```diff
@@ -20,15 +17,10 @@
     private Long id;
-    @NotNull
-    @FutureOrPresent(message = ValidationMessages.START_DATA)
     @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss")
```
Данную валидацию надо убрать

KirPyt commented on server/src/main/java/ru/practicum/shareit/booking/dto/BookingCreationDto.java:
```diff
@@ -20,15 +17,10 @@
     private LocalDateTime start;
-    @NotNull
-    @FutureOrPresent(message = ValidationMessages.END_DATA)
     @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss")
```
И данную валидацию тоже

KirPyt commented on server/src/main/java/ru/practicum/shareit/item/controller/ItemController.java:
```diff
@@ -9,7 +9,6 @@
-import javax.validation.Valid;
 import javax.validation.constraints.Min;
 import java.util.List;
 
```
Из данного класса надо убрать `@Validated` и аннотации в параметрах методов контроллеров, которые валидируют параметры (`@Min(0)` и `@Min(1)`)

KirPyt commented on server/src/main/java/ru/practicum/shareit/item/dto/CommentDto.java:
```diff
@@ -17,7 +15,6 @@ public class CommentDto {
-    @NotBlank(message = ValidationMessages.EMPTY_TEXT)
     private String text;
     private String authorName;
     @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS")
```
Данную аннотацию надо убрать

KirPyt commented on server/src/main/java/ru/practicum/shareit/item/dto/CommentDto.java:
```diff
@@ -17,7 +15,6 @@ public class CommentDto {
     private Long id;
     @JsonIgnore
```
Данную аннотацию надо убрать

KirPyt commented on server/src/main/java/ru/practicum/shareit/request/controller/ItemRequestController.java:
```diff
@@ -8,7 +8,6 @@
-import javax.validation.Valid;
 import javax.validation.constraints.Min;
 import java.util.List;
 
```
Из данного класса надо убрать `@Validated` и аннотации в параметрах методов контроллеров, которые валидируют параметры (`@Min(0)` и `@Min(1)`)

KirPyt commented on server/src/main/java/ru/practicum/shareit/request/dto/AddItemRequestDto.java:
```diff
@@ -23,9 +20,7 @@
-    @NotBlank(message = ValidationMessages.EMPTY_DESCRIPTION)
     private String description;
-    @PastOrPresent(message = ValidationMessages.REQUEST_CREATED_DATA)
     @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS")
```
Данную аннотацию надо убрать

KirPyt reviewed (commented):
Работа отличная, только есть некоторые мелкие недочёты по разделению валидации - в сервере валидации быть не должно, надо почистить соответствующие классы.

Если будут вопросы, то пишите в Пачку

LeonovichMariia committed:
fix: убран неиспользуемый код. Аннотация @Min для параметра from заменена на @PositiveOrZero
delete: удалена аннотация @Validated в классе UserController. В файле pom.xml в server удалена зависимость spring-boot-starter-validation. В server в классах BookingController, ItemController и ItemRequestController удалены аннотации валидации. Удалены аннотации @JsonFormat и @JsonIgnore из Dto-классов в server.

KirPyt commented on gateway/src/main/java/ru/practicum/shareit/item/dto/CommentDto.java:
```diff
@@ -1,25 +1,18 @@
+@NoArgsConstructor
 @AllArgsConstructor
 public class CommentDto {
-    private Long id;
```
А почему Вы удалили поля из данного класса?

KirPyt commented on gateway/src/main/java/ru/practicum/shareit/item/dto/ItemDto.java:
```diff
@@ -1,30 +1,24 @@
     private String description;
     @NotNull(message = ValidationMessages.AVAILABLE_NULL)
     private Boolean available;
-    private Long ownerId;
```
Почему из этого класса тоже удалены некоторые поля?

KirPyt commented on gateway/src/main/java/ru/practicum/shareit/user/dto/UserDto.java:
```diff
@@ -1,16 +1,19 @@
+@NoArgsConstructor
+@AllArgsConstructor
 public class UserDto {
-    private Long id;
```
Почему данное поле удалено?

KirPyt reviewed (commented):
Указал на некоторые спорные моменты - Вы удалили некоторые поля из ДТО-шек вместе с валидацией в gateway, стоит вернуть поля и валидацию. Валидации не должно быть только в сервере и в серверных ДТО-шках. В gateway валидация должна быть

LeonovichMariia committed:
fix: добавлены поля и аннотации в DTO-классы в gateway.

KirPyt commented on gateway/src/main/java/ru/practicum/shareit/item/dto/ItemDto.java:
```diff
@@ -24,7 +23,4 @@ public class ItemDto {
     private Boolean available;
     private Long ownerId;
     private Long requestId;
-    private List<CommentDto> comments;
```
Поля всё ещё удалены

KirPyt reviewed (commented):
Отлично!
Оставил небольшой комментарий к работе

Ваш проект зачтён

LeonovichMariia committed:
fix: добавлены поля в ItemDto в gateway.
